src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -193,6 +205,65 @@ class PremisesController(
     return ResponseEntity.ok(cancellationTransformer.transformJpaToApi(cancellation))
   }
 
+  override fun premisesPremisesIdBookingsBookingIdDeparturesPost(
+    premisesId: UUID,
+    bookingId: UUID,
+    body: NewDeparture
+  ): ResponseEntity<Departure> {
+    val premises = premisesService.getPremises(premisesId)
+      ?: throw NotFoundProblem(premisesId, "Premises")
+
+    val booking = bookingService.getBooking(bookingId)
+      ?: throw NotFoundProblem(bookingId, "Booking")
+
+    if (booking.premises.id != premises.id) {
+      throw NotFoundProblem(bookingId, "Booking")
+    }
+
+    if (booking.arrivalDate.toLocalDateTime().isAfter(body.dateTime)) {
+      throw BadRequestProblem(mapOf("dateTime" to "Must be after the Booking's arrival date (${booking.arrivalDate})"))
+    }
+
+    if (booking.departure != null) {
+      throw BadRequestProblem(errorDetail = "This Booking already has a Departure set")
+    }
+
+    val validationIssues = mutableMapOf<String, String>()
+
+    val reason = departureReasonRepository.findByIdOrNull(body.reasonId)
+    if (reason == null) {
+      validationIssues["reasonId"] = "Reason does not exist"
+    }
+
+    val moveOnCategory = moveOnCategoryRepository.findByIdOrNull(body.moveOnCategoryId)
+    if (reason == null) {
+      validationIssues["moveOnCategoryId"] = "Move on Category does not exist"
+    }
+
+    val destinationProvider = destinationProviderRepository.findByIdOrNull(body.destinationProviderId)
+    if (destinationProvider == null) {
+      validationIssues["destinationProviderId"] = "Destination Provider does not exist"
+    }
+
+    if (validationIssues.any()) {
+      throw BadRequestProblem(validationIssues)
+    }
```

Comment:
Just wondering if this can be broken out into the `createDeparture` service method, or something like `validateDeparture`. This can then be unit tested. I'm generally a proponent of the "Skinny Controller" approach, and I tend to get a bit nervous when I see too much logic in controllers

Reply:
Will tackle this alongside the existing endpoints in a follow up PR